app/models/micropost.rb

```diff
@@ -1,8 +1,16 @@
 class Micropost < ApplicationRecord
   belongs_to :user
   has_one_attached :image
+
   scope :newest, ->{order created_at: :desc}
   scope :by_user_id, ->(id){where user_id: id}
+  scope :feed_for, lambda {|id|
+    following_ids_query = "SELECT followed_id FROM relationships
+                           WHERE follower_id = :user_id"
+    where "user_id IN (#{following_ids_query}) OR user_id = :user_id",
+          user_id: id
```

Comment:
```suggestion
    following_ids_query = "SELECT followed_id FROM relationships
                           WHERE follower_id = :user_id"
    where "user_id IN (#{following_ids_query}) OR user_id = :user_id",
          user_id: id
```
code này đầu tiên, em đang tìm với file user id thì mình chỉ cần gộp param cho user_id thôi, cần j dùng OR
Thứ 2 nên ko dùng or thì chị thấy nó đã code scope by_user_id rồi nè
để dùng đc em tìm cách lấy dc data của followed_id là dc nè 

suy nghĩ xem, chị tim em mần dc :D